docs/CONNECTOR_PROTOCOL.md

```diff
@@ -277,6 +277,10 @@ In addition to the connector index `.elastic-connectors`, we have an additional
       service_type: string;   -> Service type of the connector
     }
   ];
+  checkpoint: {
+    data: object; -> free-form checkpoint data for the job that can be used to restart the crashed job
+    restart_attempts: number; -> number of times the job was attempted to continue from the checkpoint
```

Comment:
I wonder how you plan to use `restart_attempts` in practice.

When connector resumes a job, it will fail the job (by marking the status as `error`) after reaching the error threshold (depending on the resilience configuration). Will you have a separate threshold, e.g. `max_resume_tries`? Even so, I don't feel `restart_attempts` is necessary here, as once you reached the threshold, you will mark the job as `error`, instead of updating `restart_attempts`.

Reply:
At this point the way the job gets to `suspended` state when connector service has to terminate during job execution. It will happen, for example, when connector service has to restart due to high memory consumption, or the process gets an OOM signal. Or the host restarts.

Since the host process for the job processing will be killed, there's no way to know how many times the suspended job was picked up and restarted without storing this information somewhere - in this case in `restart_attempts`.

So resilient errors and job recover from host crashes are different topics, at this point not connected at all.

Reply:
Now I got how is `restart_attempts` used. But I don't think they need to be grouped together. Not a strong opinion